runtime-modules/content/src/lib.rs

```diff
@@ -1024,6 +1063,21 @@ decl_module! {
                 &channel,
             )?;
 
+            // verify that upload params are valid
+            if let Some(upload_assets) = params.assets_to_upload.as_ref() {
+                let dyn_bag = DynamicBagIdType::<T::MemberId, T::ChannelId>::Channel(channel_id.clone());
+                let bag_id = BagIdType::from(dyn_bag.clone());
+
+                let upload_params = UploadParametersRecord {
+                    bag_id: bag_id.clone(),
+                    object_creation_list: upload_assets.object_creation_list.clone(),
+                    deletion_prize_source_account_id: sender.clone(),
+                    expected_data_size_fee: upload_assets.expected_data_size_fee,
+                };
+
+                Storage::<T>::can_upload_data_objects(&upload_params)?;
```

Comment:
`can_delete_data_objects` check is still missing

Reply:
from the storage pallet
``` rust
    fn can_delete_data_objects(
        bag_id: &BagId<T>,
        objects: &BTreeSet<T::DataObjectId>,
    ) -> DispatchResult {
        Self::validate_delete_data_objects_params(bag_id, objects).map(|_| ())
    }

    fn delete_data_objects(
        deletion_prize_account_id: T::AccountId,
        bag_id: BagId<T>,
        objects: BTreeSet<T::DataObjectId>,
    ) -> DispatchResult {
        let bag = Self::ensure_bag_exists(&bag_id)?;

        let bag_change = Self::validate_delete_data_objects_params(&bag_id, &objects)?;
...
```
check is still performed beforehand. So if `validate_delete_data_objects_params` fails then all the subsequent operations fails, and the storage is not modified.
What I want to ensure is that it cannot be that the first storage operation succeeds and the second operation fails. Hence the check `can_upload_data_parameters_beforehand`